src/many-ledger/src/migration/tokens.rs

```diff
@@ -0,0 +1,190 @@
+}
+
+fn migrate_token(
+    storage: &mut merk::Merk,
+    extra: &HashMap<String, Value>,
+) -> Result<(), ManyError> {
+    // Make sure we have all the parameters we need for this migration
+    let params = [
+        "token_identity",
+        "token_next_subresource",
+        "symbol",
+        "symbol_name",
+        "symbol_decimals",
+        "symbol_total",
+        "symbol_circulating",
+        "symbol_maximum",
+        "symbol_owner",
+    ];
+    for param in params {
+        if !extra.contains_key(param) {
+            return Err(ManyError::unknown(format!(
+                "Missing extra parameter '{param}' for Token Migration"
+            )));
+        }
+    }
+
+    let token_identity: String = serde_json::from_value(extra["token_identity"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let token_identity = Address::from_str(&token_identity)?;
+
+    let token_next_subresource: u32 =
+        serde_json::from_value(extra["token_next_subresource"].clone())
+            .map_err(ManyError::deserialization_error)?;
+
+    let symbol: String = serde_json::from_value(extra["symbol"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let symbol = Symbol::from_str(&symbol)?;
+
+    // Get symbol list from DB
+    let symbol_and_ticker_enc = storage
+        .get(SYMBOLS_ROOT.as_bytes())
+        .map_err(error::storage_get_failed)?
+        .ok_or_else(|| error::storage_key_not_found(SYMBOLS_ROOT))?;
+
+    let symbol_and_ticker: BTreeMap<Address, String> =
+        minicbor::decode(&symbol_and_ticker_enc).map_err(ManyError::deserialization_error)?;
+
+    // Get the symbol ticker from symbol list
+    let ticker = symbol_and_ticker
+        .get(&symbol)
+        .ok_or_else(|| ManyError::unknown(format!("Symbol {symbol} not found in DB")))
+        .cloned()?;
+
+    let owner: Option<Address> = serde_json::from_value(extra["symbol_owner"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let maximum: Option<TokenAmount> = serde_json::from_value(extra["symbol_maximum"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let name: String = serde_json::from_value(extra["symbol_name"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let decimals: u64 = serde_json::from_value(extra["symbol_decimals"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let total: TokenAmount = serde_json::from_value(extra["symbol_total"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let circulating: TokenAmount = serde_json::from_value(extra["symbol_circulating"].clone())
+        .map_err(ManyError::deserialization_error)?;
+    let info = TokenInfo {
+        symbol,
+        summary: TokenInfoSummary {
+            name,
+            ticker,
+            decimals,
+        },
+        supply: TokenInfoSupply {
+            total,
+            circulating,
+            maximum,
+        },
+        owner,
+    };
```

Comment:
I tried to make it easier to follow and less repetitive. Best would really be to have a type that directly deserialize, but 🤷 

```suggestion
    let info = (move || {
        Ok::<_, serde_json::Error>(TokenInfo {
            symbol,
            summary: TokenInfoSummary {
                name: serde_json::from_value(extra["symbol_name"].clone())?,
                ticker,
                decimals: serde_json::from_value(extra["symbol_decimals"].clone())?,
            },
            supply: TokenInfoSupply {
                total: serde_json::from_value(extra["symbol_total"].clone())?,
                circulating: serde_json::from_value(extra["symbol_circulating"].clone())?,
                maximum: serde_json::from_value(extra["symbol_maximum"].clone())?,
            },
            owner: serde_json::from_value(extra["symbol_owner"].clone())?,
        })
    })()
    .map_err(ManyError::deserialization_error)?;
```

Reply:
That's cute!